client-test/tests/client.rs

```diff
@@ -555,7 +557,11 @@ async fn test_slot_subscription_async() {
         ));
         let (trigger, pubsub_service) =
             PubSubService::new(PubSubConfig::default(), &subscriptions, pubsub_addr);
-        sleep(Duration::from_millis(100));
+
+        if !check_server_is_ready(&pubsub_addr, 10, Duration::from_millis(100)) {
```

Comment:
wdyt about moving the `eprintln!("server initialized timeout");` and the `if` into `check_server_is_ready()` to remove the duplication in each individual test.

Also instead of `eprintln!`, just `panic!()`, as the test is unlikely to pass on timeout. Better to fail quick

Reply:
sounds good! updated at [cd03261](https://github.com/solana-labs/solana/pull/29255/commits/cd032613e55e9a4f6e58232c87b34b4b7346db57)

Reply:
btw, I pushed a new commit [08cfdc8](https://github.com/solana-labs/solana/pull/29255/commits/08cfdc871d67f5d469075eeabfaae2c13927ba10)
I used processed level commitment to test `test_rpc_client`. It can reduce the test time from 23.44s to 13.52s (in my dedicated server 😈 )